talker/client.py

```diff
@@ -216,6 +218,16 @@ def is_done(self, cmds):
         """
         return len(list(filter(lambda i: i is None, self.poll(cmds)))) == 0
 
+    def is_alive(self, host_id, name='is_alive', timeout=5):
+        """
+        Do a naive check to see if the agent is responsive.
+        :return: True if we get ack, False otherwise
+        If the function returns False it doesn't mean for sure that the agent/machine is down
+        """
+        cmd = self.run(host_id, 'true', name=name, timeout=timeout, ack_timeout=5)
+        cmd.wait()
```

Comment:
When will this wait return?

Reply:
when ack is timed out or command is timed out.